pkg/services/grafana-apiserver/utils/uids.go

```diff
@@ -4,19 +4,25 @@ import (
 	"crypto/sha256"
 	"encoding/base64"
 	"strings"
+	"unicode"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 // Create a stable UID that will be unique across a multi-tenant cluster
-func CalculateClusterWideUID(obj metav1.Object) types.UID {
+func CalculateClusterWideUID(obj metav1.Object, kind string) types.UID {
 	hasher := sha256.New()
-	hasher.Write([]byte(obj.GetResourceVersion()))
+	hasher.Write([]byte(kind))
```

Comment:
should we use group and resource here?

Reply:
I started down that path... but it required too many extra typecheck castings 🤮 and the value of this is for things that you already know what the kind is so it seemed easier to just pass in a string

Reply:
🤔  or pass in runtime.Object and then cast to`metav1.Object`

Reply:
i updated the function to take runtime.Object -- all this casting is 🤯 sometimes!